odl/deform/linearized_deform.py

```diff
@@ -0,0 +1,546 @@
+        >>> template = space.element([0, 0, 1, 0, 0])
+        >>> disp_field = disp_field_space.element([[0, 0, 0, -0.2, 0]])
+        >>> vector_field = disp_field_space.element([[1, 1, 1, 2, 1]])
+        >>> op = LinDeformFixedTemplDeriv(template, disp_field)
+        >>> op(vector_field)
+        uniform_discr(0.0, 1.0, 5).element([0.0, 5.0, -5.0, -10.0, 0.0])
+        """
+        if displacement[0] not in template.space:
+            raise TypeError('`displacement[0]` {!r} not an element of'
+                            '`template.space`'.format(displacement[0],
+                                                      template.space))
+
+        Operator.__init__(self, displacement.space, template.space,
+                          linear=True)
+
+        self.template = template
+        self.displacement = displacement
+
+    def _call(self, vector_field):
+        """Implementation of ``self(vector_field)``.
+
+        Parameters
+        ----------
+        vector_field: `domain` element
+            The evaluation point, i.e. a displacement field, for the
+            deformed gradient of the template.
+        """
+        if vector_field not in self.displacement.space:
+            raise TypeError('`vector_field` {!r} not an element of'
+                            '`displacement.space`'.format(
+                                vector_field, self.displacement.space))
+
+        grad = Gradient(self.range, method='forward',
+                        padding_method='symmetric')
+        grad_template = grad(self.template)
+        def_grad = deform_grad(grad_template, self.displacement)
+
+        innerProdOp = PointwiseInner(self.displacement.space, def_grad)
+
+        return innerProdOp(vector_field)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the derivative operator.
+
+        Returns
+        -------
+        adjoint : `LinDeformFixedTemplDerivAdj`
+            The adjoint of the operator.
+        """
+        return LinDeformFixedTemplDerivAdj(self.template, self.displacement)
+
+
+class LinDeformFixedTemplDerivAdj(Operator):
+
+    """Adjoint operator of the derivative operator.
+
+    This operator computes the adjoint of the derivative of the fixed
+    template linearized deformation operator for the fixed template ``I``
+    at given displacement ``v``:
+
+        J --> grad(I)(Id + v) J
+
+    where the ``J`` is the given template and
+    the ``Id`` is the identity mapping:
+
+        y --> y
+
+    and the vector field ``v`` is a displacement mapping:
+
+        y --> v(y)
+
+    Here, the ``y`` is an element in the domain.
+    """
+
+    def __init__(self, template, displacement):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        template : `DiscreteLp` element
+            Fixed template deformed by the vector field.
+        displacement: `ProductSpace` element
+            Point that the adjoint of the derivative needs
+            to be computed at.
```

Comment:
"needs to be" sounds undecided. Better: "Point at which the derivative is taken." We also do not calculate the adjoint of the derivative at a point, we calculate the derivative at a point and take the adjoint of the result. That's a difference.